core/vm/contract.go

```diff
@@ -185,26 +149,23 @@ func (c *Contract) RefundGas(gas uint64, logger *tracing.Hooks, reason tracing.G
 
 // Address returns the contracts address
 func (c *Contract) Address() common.Address {
-	return c.self.Address()
+	return c.address
 }
 
 // Value returns the contract's value (sent to it from it's caller)
 func (c *Contract) Value() *uint256.Int {
 	return c.value
 }
 
-// SetCallCode sets the code of the contract and address of the backing data
-// object
-func (c *Contract) SetCallCode(addr *common.Address, hash common.Hash, code []byte) {
+// SetCallCode sets the code of the contract,
+func (c *Contract) SetCallCode(hash common.Hash, code []byte) {
 	c.Code = code
 	c.CodeHash = hash
-	c.CodeAddr = addr
 }
 
 // SetCodeOptionalHash can be used to provide code, but it's optional to provide hash.
 // In case hash is not provided, the jumpdest analysis will not be saved to the parent context
-func (c *Contract) SetCodeOptionalHash(addr *common.Address, codeAndHash *codeAndHash) {
+func (c *Contract) SetCodeOptionalHash(codeAndHash *codeAndHash) {
```

Comment:
It looks like the hash is only set in the Create2 case, does that mean that we save the jumpdest analysis for initcode created with create2 but not with create? Can this be used to waste a bunch of memory on the node?

Reply:
Looks like this is the case, I managed to get a test from 30MB to 60MB memory consumption when doing a bunch of initcode shenanigans with create2 instead of create. It's not a real issue, but we should discuss nonetheless

Reply:
Yeah, I agree that we somehow save the init code in the jumpDests with create2.

Probably we should change the function to forcibly only set the code, to explicitly avoid caching init code?

Reply:
Yeah I think we could just set the code in `create` and get rid of the `codeAndHash` in `Create` and `Create2`